src/components/PatchworkEmbed/PatchworkEmbed.tsx

```diff
@@ -0,0 +1,190 @@
+  animationSpeedMap = defaultConfig.animationSpeedMap,
+  animationSpeed = defaultConfig.animationSpeed,
+  ...props
+}) => {
+  const viewport = useRef<any>();
+
+  const height =
+    window.innerWidth < mobileBreakpoint ? mobileHeight : desktopHeight;
+
+  // State.
+  const [isMobileFullscreen, setIsMobileFullscreen] = useState(false);
+  const [annotation, setAnnotation] = useState<Annotation>();
+
+  const getAnimationSpeed = (name: string) => {
+    if (animationSpeedMap[name]) {
+      return animationSpeedMap[name] / 1000;
+    }
+    return animationSpeed / 1000;
+  };
+
+  // Actions.
+  const onClickAnnotation = (selectedAnnotation: Annotation, bounds: any) => {
+    if (clickToClose && annotation && annotation.id === selectedAnnotation.id) {
+      return onClose();
+    }
+
+    setAnnotation(selectedAnnotation);
+    viewport.current.goToRect(
+      bounds,
+      annotationMargin,
+      getAnimationSpeed('onClick')
+    );
+  };
+  const toggleMobileFullscreen = () => {
+    setIsMobileFullscreen(current => !current);
+  };
+  const onClose = () => {
+    setAnnotation(undefined);
+
+    viewport.current.resetView(getAnimationSpeed('onClose'));
+  };
+
+  return (
+    <Bem
+      prefix="patchwork-"
+      cssClassMap={{
+        annotation: 'annotation-pin',
+      }}
+    >
+      <CanvasProvider currentCanvas={canvas.id}>
+        <AdaptiveViewport
+          isFullscreen={fullscreenEnabled ? isFullscreen : false}
+          fullViewport={fitContainer}
+          maxHeight={
+            isFullscreen || isMobileFullscreen ? window.innerHeight : height
+          }
+          setRef={(ref: any) => (viewport.current = ref)}
+        >
+          <SingleTileSource viewportController={true}>
+            <OpenSeadragonViewport>
+              <OpenSeadragonViewer
+                useMaxDimensions={true}
+                osdOptions={{
+                  visibilityRatio: 1,
+                  constrainDuringPan: true,
+                  showNavigator: false,
+                  immediateRender: false,
+                  ...osdOptions,
+                }}
+              />
+            </OpenSeadragonViewport>
+          </SingleTileSource>
+          <AnnotationCanvasRepresentation
+            ratio={0.1}
+            // @ts-ignore
+            // ratioFromMaxWidth={1000}
+            growthStyle={growthStyle as any}
+            bemModifiers={anno => ({
+              selected: annotation ? annotation.id === anno.id : false,
+            })}
+            onClickAnnotation={onClickAnnotation}
+          />
+
+          {annotation ? (
+            <AnnotationDetail
+              data-static
+              closeText={closeText}
+              annotation={annotation}
+              onClose={onClose}
+              cssClassMap={{
+                'annotation-detail': 'patchwork-annotation-detail',
+              }}
+            />
+          ) : null}
+        </AdaptiveViewport>
+      </CanvasProvider>
+    </Bem>
+  );
+};
```

Comment:
Here's the patchwork plugin, but without the manifest context. What it should mean is that you can drop it into an existing manifest context and it will pick it up. You do need to pass in the canvas though, the CanvasRangeProvider doesn't create a `CanvasContext` as it should